### PR TITLE
Fix analyze-build warnings

### LINKED ISSUE #390: analyze-build warnings
### Information

- **RTI Product**: Connext DDS
- **Version**: 6.1.0
- **Operating system**: Linux
- **Compiler**: GCC
- **Compiler version**: 7.5.0
- **Additional information**:

### What is the current behavior?

There are a few warnings during static analysis:

```plaintext
analyze-build: INFO: /home/mnunez/rticonnextdds-examples/examples/connext_dds/custom_transport/c/FileTransport.c:1504:5: warning: Dereference of null pointer (loaded from variable 'dest_address_in')
analyze-build: INFO:     NDDS_Transport_Address_copy(&sendResourceStruct->_address, dest_address_in);
analyze-build: INFO:     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
analyze-build: INFO: /home/mnunez/rti_connext_dds-6.1.0/include/ndds/transport/transport_common_impl.h:68:15: note: expanded from macro 'NDDS_Transport_Address_copy'
analyze-build: INFO:    ( *(dst) = *(src) )
analyze-build: INFO:               ^~~~~~
analyze-build: INFO: /home/mnunez/rticonnextdds-examples/examples/connext_dds/custom_transport/c/FileTransport.c:1553:14: warning: Null pointer passed as an argument to a 'nonnull' parameter
analyze-build: INFO:         || (!NDDS_Transport_Address_is_equal(
analyze-build: INFO:              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
analyze-build: INFO: /home/mnunez/rti_connext_dds-6.1.0/include/ndds/transport/transport_common_impl.h:63:7: note: expanded from macro 'NDDS_Transport_Address_is_equal'
analyze-build: INFO:    ( !RTIOsapiMemory_compare(l, r, sizeof(NDDS_Transport_Address_t)) )
analyze-build: INFO:       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
analyze-build: INFO: /home/mnunez/rti_connext_dds-6.1.0/include/ndds/osapi/osapi_bufferUtils_impl.h:44:30: note: expanded from macro 'RTIOsapiMemory_compare'
analyze-build: INFO:         (((size) == 0) ? 0 : memcmp(l, r, (size_t) (size)))
analyze-build: INFO:                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
analyze-build: INFO: 2 warnings generated.
analyze-build: INFO: /home/mnunez/rticonnextdds-examples/examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx:162:45: warning: Division by zero
analyze-build: INFO:     std::cout << "Avg. pixel: (" << red_sum / pixel_count << ", "
analyze-build: INFO:                                     ~~~~~~~~^~~~~~~~~~~~~
analyze-build: INFO: 1 warning generated.
analyze-build: INFO: /home/mnunez/rticonnextdds-examples/examples/connext_dds/logging_config/c/logging_publisher.c:94:27: warning: Value stored to 'send_period' during its initialization is never read
analyze-build: INFO:     struct DDS_Duration_t send_period = {4,0};
analyze-build: INFO:                           ^~~~~~~~~~~   ~~~~~
analyze-build: INFO: 1 warning generated.
```

### Steps to reproduce the issue

Run `resources/ci_cd/linux_static_analysis.py`.

### PR DESCRIPTION
**Pull request by ManuelJNunez**
:white_check_mark: I have updated the documentation accordingly.
:white_check_mark: I have read the CONTRIBUTING document.

### Summary

I fixed the warnings of `analyze-build` (take a look of the issue closed bellow).

### Details and comments

close #390

### CODE REVIEW

**Review by iblancasa — APPROVED** [review]

**Review by lulivi — APPROVED** [review]

File: examples/connext_dds/custom_transport/c/FileTransport.c [hunk 1]
```diff
@@ -1497,6 +1497,11 @@ static RTI_INT32 NDDS_Transport_FILE_create_sendresource_srEA(
         return 0;
     }
 
+    /* Check if dest_address_in is null */
```
Comment (fgaranda) [inline examples/connext_dds/custom_transport/c/FileTransport.c:1500]: Move precondition check before you use the variable. I guess you'd need to do this check on line 1484 right?

**Review by fgaranda — CHANGES_REQUESTED** [review]

File: examples/connext_dds/custom_transport/c/FileTransport.c [hunk 2]
```diff
@@ -1546,6 +1551,11 @@ static RTI_INT32 NDDS_Transport_FILE_share_sendresource_srEA(
...
```
Comment (fgaranda) [inline examples/connext_dds/custom_transport/c/FileTransport.c:1554]: Same here

**Comment by ManuelJNunez** [comment]
Travis has the service degraded today...

**Review by lulivi — APPROVED** [review]

File: examples/connext_dds/custom_transport/c/FileTransport.c [hunk 3]
```diff
@@ -1482,6 +1482,11 @@ static RTI_INT32 NDDS_Transport_FILE_create_sendresource_srEA(
 
     *sendresource_out = NULL;
 
+    /* Check if dest_address_in is null */
+    if (dest_address_in == NULL) {
+        return 0;
```
Comment (angelrti) [inline examples/connext_dds/custom_transport/c/FileTransport.c:1487]: this is something that is tested in the function `NDDS_Transport_Address_is_multicast` so it is not really needed. If this throws a warning and we need to check `!= NULL`, we should add a message error

File: examples/connext_dds/custom_transport/c/FileTransport.c [hunk 4]
```diff
@@ -1541,6 +1546,11 @@ static RTI_INT32 NDDS_Transport_FILE_share_sendresource_srEA(
     struct NDDS_Transport_SendResource_FILE *sendResourceStruct =
             (struct NDDS_Transport_SendResource_FILE *) *sendresource_in;
 
+    /* Check if dest_address_in is null */
+    if (dest_address_in == NULL) {
```
Comment (angelrti) [inline examples/connext_dds/custom_transport/c/FileTransport.c:1550]: same here

File: examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx [hunk 5]
```diff
@@ -146,6 +146,10 @@ bool print_average_pixel_fast(const CameraImage &sample)
 
     unsigned int pixel_count = pixels.element_count();
 
+    if (pixel_count == 0) {
+        return false;
```
Comment (angelrti) [inline examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx:150]: log a message if this happens

**Review by angelrti — CHANGES_REQUESTED** [review]

File: examples/connext_dds/custom_transport/c/FileTransport.c (hunk 3 above)
Comment (ManuelJNunez) [reply examples/connext_dds/custom_transport/c/FileTransport.c:1487]: Yes, but the static analyzer threw a warning, so I fixed it.

File: examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx [hunk 6]
```diff
@@ -146,6 +146,11 @@ bool print_average_pixel_fast(const CameraImage &sample)
 
     unsigned int pixel_count = pixels.element_count();
 
+    if (pixel_count == 0) {
+        print("The number of pixels should be greater than 0.");
```
Comment (lulivi) [inline examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx:150]: Does `print` exist in C++?

**Review by lulivi — CHANGES_REQUESTED** [review]

**Review by lulivi — APPROVED** [review]

File: examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx [hunk 7]
```diff
@@ -146,6 +146,11 @@ bool print_average_pixel_fast(const CameraImage &sample)
 
     unsigned int pixel_count = pixels.element_count();
 
+    if (pixel_count == 0) {
+        printf("The number of pixels should be greater than 0.");
```
Comment (angelrti) [inline examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx:150]: use `std::cout` instead
probably we can also change the message to "WARNING: the pixels count is 0"

**Review by angelrti — CHANGES_REQUESTED** [review]

**Review by angelrti — APPROVED** [review]

**Pull request merged**